test/studies/590o/wk3/05argQueries.chpl

```diff
@@ -15,7 +15,7 @@ writeln();
  */
 proc printShifts(x: int(?width)) {
   writeln("the value ", x, "...");
-  for i in 1..width do
+  for i in 1..(width - 1) do
```

Comment:
now I am a bit confused about your intentions since you changed this test. I believe I should be able to shift a 64 bit value 64 times (until it is zeroed out basically)?

Reply:
That is generally considered to be undefined behavior by C compilers. Chapel could define such semantics, but it would have to be implemented in the form of:

```chapel
if b >= numBits(a) then return 0;
```

For well defined, consistent, cross platform behavior. The alternative is to have checks and just turn them off (what we do here).

Reply:
Right, AFAIK, C compilers assume that shifts could be implemented in hardware that does a modulus operation on the shift amount, e.g. `shiftAmt % 64`. So while many of us would like `myUint >> 64` to result in 0, that's not actually given by the C standard.

We could consider allowing it in the future (without breaking any programs).

Reply:
I understand, I did not know that. There is no change needed, then.